R/metric-selection.R

```diff
@@ -115,10 +115,10 @@ choose_eval_time <- function(x, metric, eval_time = NULL, quietly = FALSE, call
 
   # If we don't need an eval time but one is passed:
   if (!dyn_metric & !is.null(eval_time) & !quietly) {
-    cli::cli_warn("An evaluation time is only required when a dynamic
-                   metric is selected (and {.arg eval_time} will thus be
-                   ignored).",
-                  call = call)
+    cli::cli_warn(
```

Comment:
I think that this warning might be triggered when there is a metric set that contains dynamic and integrated metrics. I'm ok with the new version but, just for clarity, the old one was written that obliquely suggests that there would be multiple metrics in the metric set (and a dynamic metric was not selected).

Reply:
> that there would be multiple metrics in the metric set (and a dynamic metric was not selected).
Do you mean that a metrics set was used to create the `tune_results` object or that the user specified multiple metrics in the function triggering the warning?

We can arrive at this warning from `choose_eval_time()` via `show_best()`, `select_best()`, or `fit_best()` i.e., situations where we may have used a (mixed) metric set for the resampling/tuning but are now picking a single metric (and possibly eval time). 

If we feed multiple metrics to e.g. `show_best()`, it selects a single metric first and warns about it, before the warning about eval_time is triggered. 

I think neither wording makes it very clear that you could have used multiple metrics when creating the `tune_results` object but I'm also not sure that this is super important info? (Possibly the reason it was somewhat oblique to start with?). The wording in this PR makes it consistent with similar messages so that's my main reason for having worded it the way I did :) 

``` r
library(tidymodels)
library(censored)
#> Loading required package: survival

set.seed(1)
sim_dat <- prodlim::SimSurv(500) %>%
  mutate(event_time = Surv(time, event)) %>%
  select(event_time, X1, X2)
sim_rs <- vfold_cv(sim_dat)

mod_spec <-
  bag_tree() %>%
  set_mode("censored regression")

mix_mtrc  <- metric_set(brier_survival, brier_survival_integrated, concordance_survival)

set.seed(2193)
rs_mixed_res <-
  mod_spec %>%
  fit_resamples(
    event_time ~ X1 + X2,
    resamples = sim_rs,
    metrics = mix_mtrc,
    eval_time = c(10, 1, 5, 15)
  )

show_best(rs_mixed_res, metric = "brier_survival_integrated", eval_time = 5)
#> Warning in show_best(rs_mixed_res, metric = "brier_survival_integrated", :
#> `eval_time` is only used for dynamic survival metrics.
#> # A tibble: 1 × 7
#>   .metric                   .estimator .eval_time  mean     n std_err .config   
#>   <chr>                     <chr>           <dbl> <dbl> <int>   <dbl> <chr>     
#> 1 brier_survival_integrated standard           NA 0.123    10  0.0112 Preproces…

show_best(rs_mixed_res, metric = c("brier_survival_integrated", "brier_survival"), eval_time = 5)
#> Warning in show_best(rs_mixed_res, metric = c("brier_survival_integrated", : 2
#> metrics were given; "brier_survival_integrated" will be used.
#> Warning in show_best(rs_mixed_res, metric = c("brier_survival_integrated", :
#> `eval_time` is only used for dynamic survival metrics.
#> # A tibble: 1 × 7
#>   .metric                   .estimator .eval_time  mean     n std_err .config   
#>   <chr>                     <chr>           <dbl> <dbl> <int>   <dbl> <chr>     
#> 1 brier_survival_integrated standard           NA 0.123    10  0.0112 Preproces…
```

<sup>Created on 2024-02-19 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>